### PR TITLE
[doc] Updated documentations for implicit type casting rules

### PR DESCRIPTION
**Pull request by jim19930609**
Related issue = #4793

### CODE REVIEW

**Review by strongoier — COMMENTED** [review]
As our rules are different from C now, shall we re-write this subsection by directly showing our own rules? IMO we have three rules now (high priority -> low priority):
- int + float
- low precision bits + high precision bits
- int + unsigned

BTW we should also exclude some binary operations here (e.g. shift operations in #4884).

File: docs/lang/articles/basic/type.md [hunk 1]
```diff
@@ -100,10 +100,33 @@ As a rule of thumb, implicit type casting is a major source of bugs. And Taichi
 
 #### Implicit type casting in binary operations
 
-Taichi follows the [implicit conversion rules](https://en.cppreference.com/w/c/language/conversion) for the C programming language and implicitly casts operands in a [binary operation](https://en.wikipedia.org/wiki/Binary_operation) into a *common type* if the operation involves different data types. Following are two most straightforward rules for determining the common type in a binary operation:
+Taichi implements its own implicit type casting rules for binary operations, which is slightly different from that for C programming language: [implicit conversion rules](https://en.cppreference.com/w/c/language/conversion) and [binary operation](https://en.wikipedia.org/wiki/Binary_operation).
```
Comment (strongoier) [inline docs/lang/articles/basic/type.md:103]:
```suggestion
Taichi implements its own implicit type casting rules for binary operations, which are slightly different from [those for the C programming language](https://en.cppreference.com/w/c/language/conversion).
```

File: docs/lang/articles/basic/type.md [hunk 2]
```diff
@@ -100,10 +100,33 @@ As a rule of thumb, implicit type casting is a major source of bugs. And Taichi
...
+
+1. integral OP floating_point -> floating_point
+- `i32 + f32 -> f32`
+- `i16 + f16 -> f16`
+
+2. low_precision_bits OP high_precision_bits -> high_precision_bits
+- `i16 + i32 -> i32`
+- `u8 + u16 -> u16`
+
+3. signed OP unsigned -> signed
```
Comment (strongoier) [inline docs/lang/articles/basic/type.md:115]: The result should be unsigned..

Comment (jim19930609) [reply docs/lang/articles/basic/type.md:115]: Many thanks!

File: docs/lang/articles/basic/type.md (hunk 1 above)
Comment (jim19930609) [reply docs/lang/articles/basic/type.md]: Done. Thanks!

**Review by wyushun — COMMENTED** [review]
Thumbs up for this doc!

**Review by strongoier — APPROVED** [review]
LGTM! @Vissidarte-Herman would you take a look at the writing style? Thanks!

**Pull request merged**